### PR TITLE
[Chart & Dashboard] improve listview filter ui and add expandable list support

### PR DESCRIPTION
**Pull request by nytai**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

- fixes CSS for dropdown select
- adds search input submit and clear actions (+icons)
- adds expandable list for owners in Dashboard list view
- adds row selected CSS

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

Screen Shot 2020-04-24 at 3 40 53 PM
Screen Shot 2020-04-24 at 3 46 22 PM
Screen Shot 2020-04-24 at 3 44 29 PM
Screen Shot 2020-04-24 at 3 44 47 PM
Screen Shot 2020-04-26 at 2 09 40 PM

### TEST PLAN

- unit tested
- manual testing

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [x] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@rusackas @mistercrunch @dpgaspar

### CODE REVIEW

File: superset-frontend/src/components/ExpandableList.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  return arr.slice(1).reduce((xs, x) => xs.concat([sep, x]), [arr[0]]);
+}
+
+const ActionButton: React.FunctionComponent<{ onClick: () => void }> = ({
+  onClick,
+  children,
+}) => (
+  <Button
+    bsStyle="link"
+    css={css`
```
Comment (rusackas) [inline superset-frontend/src/components/ExpandableList.tsx:43]: These little ad hoc css tweaks, in the long run, will make me nervous. At some point, we'll need to make the super-awesome `SupersetButton` that does everything we need it to with props, so we don't need to pepper overrides around the codebase. Not a big concern for this PR, but pointing out some hopes and dreams :)

File: superset-frontend/src/components/ListView/ListViewStyles.less [hunk 2]
```diff
@@ -61,6 +64,10 @@
     margin: 0 8px;
   }
 
+  .table-row-selected {
+    background-color: @gray-bg;
```
Comment (rusackas) [inline superset-frontend/src/components/ListView/ListViewStyles.less:68]: Thanks for using the variable! 🥇

File: superset-frontend/src/components/SearchInput.tsx [hunk 3]
```diff
@@ -17,13 +17,95 @@
...
+  onSubmit,
+  onClear,
+  placeholder = 'Search',
+  value,
+  onChange,
+}: Props) {
+  return (
+    <div
+      css={css`
+        position: relative;
```
Comment (rusackas) [inline superset-frontend/src/components/SearchInput.tsx:82]: It's not relative by default? Kind of odd. I also wonder if this makes more sense here, or in the SASS of the `styled.span` above, as a direct descendant selector, `& > div {...}`

**Review by rusackas — APPROVED** [review]
LGTM, Minor comments noted

File: superset-frontend/src/components/ExpandableList.tsx (hunk 1 above)
Comment (nytai) [reply superset-frontend/src/components/ExpandableList.tsx:43]: yea we need a `<Button as="Link"/>` or something similar. I'll move this to `src/components`

**Review by rusackas — APPROVED** [review]

**Pull request merged**